File: src/main/kotlin/id/sekawan/point/util/mylib/HttpException.java
package id.sekawan.point.util.mylib;

public class HttpException extends RuntimeException {
    private int httpStatusCode;

    public HttpException(String message, int httpStatusCode) {
        super(message);
        this.httpStatusCode = httpStatusCode;
    }

    public int getHttpStatusCode() {
        return this.httpStatusCode;
    }
}

File: src/main/kotlin/id/sekawan/point/util/HttpException.kt
package id.sekawan.point.util

class HttpException(message: String, httpStatusCode: Int) : RuntimeException(message) {
    private var errorCode: Int = 500
    fun getErrorCode():Int { return errorCode }

    init {
        RuntimeException(message)
        errorCode = httpStatusCode
    }
}